src/screen-tracker.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import type { Middleware } from 'redux';
+
+import type {
+  ReducerState,
+  NavStateSelector,
+  OnScreenChangeCallback,
+} from './types';
+
+import { NavigationActions } from "react-navigation";
+
+function getCurrentRouteName(navigationState: ReducerState) {
+  if (!navigationState) return null;
+
+  const route = navigationState.routes[navigationState.index];
+
+  // dive into nested navigators
+  if (route.routes) return getCurrentRouteName(route);
```

Comment:
@Ashoat this is another situation where the type is failing 😢 

```
[flow] Cannot call `getCurrentRouteName` with `route` bound to `navigationState` because property `routes` is missing in `NavigationLeafRoute` [1] but exists in `NavigationState` [2]. (References: [1] [2])
```